ansible/roles/dev-desktop/files/team_login/src/main.rs

```diff
@@ -0,0 +1,105 @@
+const TEAM_URL: &str = "https://team-api.infra.rust-lang.org/v1/teams/all.json";
+
+use miniserde::Deserialize;
+use std::collections::HashSet;
+use std::process::Command;
+use std::process::Output;
+
+#[derive(Deserialize)]
+struct All {
+    members: Vec<Person>,
+}
+
+#[derive(Deserialize)]
+struct Person {
+    github: String,
+}
+
+fn cmd(cmd: &str, args: &[&str]) -> std::io::Result<Output> {
+    Command::new(cmd).args(args).output()
+}
+
+const KEY_DIR: &str = "/etc/ssh/authorized_keys/";
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let mut handle = curl::easy::Easy::new();
+    handle.url(TEAM_URL).unwrap();
+    let mut data = Vec::new();
+    {
+        let mut transfer = handle.transfer();
+        transfer
+            .write_function(|new_data| {
+                data.extend_from_slice(new_data);
+                Ok(new_data.len())
+            })
+            .unwrap();
+        transfer.perform().unwrap();
+    }
+
+    let data = String::from_utf8(data).unwrap();
+    let all: All = miniserde::json::from_str(&data).unwrap();
+    let mut users = HashSet::new();
+    for person in all.members {
+        users.insert(person.github.clone());
+        // Get the keys the user added to their github account.
+        // Do this every time the cronjob runs so that they get their new keys if necessary.
+        let mut keys = Vec::new();
+        handle
+            .url(&format!("https://github.com/{}.keys", person.github))
```

Comment:
This is going to result in a lot of requests every 5 minutes, do we know what the rate limit on the endpoint is?